Source/WebKit/UIProcess/API/C/WKNavigationResponseRef.cpp

```diff
@@ -43,3 +43,8 @@ WKURLResponseRef WKNavigationResponseCopyResponse(WKNavigationResponseRef respon
 {
     return WebKit::toAPI(API::URLResponse::create(WebKit::toImpl(response)->response()).leakRef());
 }
+
+WKFrameInfoRef WKNavigationResponseGetFrameInfoRef(WKNavigationResponseRef response)
```

Comment:
I think WK2 API is using `Copy` to return a ref counted object instead of `Get` like `WKNavigationActionCopyTargetFrameInfo`.
I think you should name `WKNavigationResponseCopyFrameInfoRef` and the client has to deref.

Reply:
NavigationResponse does not have information to create FrameInfo. Also, NavigationResponse::frame() returns FrameInfo& instead of Ref<FrameInfo>&. Therefore, it is not possible to return a ref counted object. What should I do?

https://github.com/WebKit/WebKit/blob/3fe5ba4547a3ece6a1ab059a583e183b06e6da54/Source/WebKit/UIProcess/API/APINavigationResponse.h#L44
https://github.com/WebKit/WebKit/blob/3fe5ba4547a3ece6a1ab059a583e183b06e6da54/Source/WebKit/UIProcess/API/APIFrameInfo.h#L49

Reply:
Removed `WKNavigationResponseGetFrameInfoRef` and created `WKNavigationResponseIsMainFrame` instead. How do you think?

Reply:
You can create a `Ref` object from a raw reference. And, it sounds a good idea to change the return type of `NavigationResponse::frame()` to `Ref&` to reduce the usage of raw references.
I prefer `WKNavigationResponseCopyFrameInfoRef`.

Reply:
I couldn't find a function that returns a Ref& under the API directory. So I don't change NavigationResponse::frame().

Reply:
Fixed